类型检查失败: scheduler.test.ts - MockInstance 类型不兼容

## 测试输出错误信息

```
src/schedule/scheduler.test.ts(102,5): error TS2322: Type 'MockInstance<(chatId: string, callbacks: PilotCallbacks, options?: AgentCreateOptions) => ChatAgent>' is not assignable to type 'MockInstance<(this: unknown, ...args: unknown[]) => unknown>'.
  Types of property 'mock' are incompatible.
    Type 'MockContext<(chatId: string, callbacks: PilotCallbacks, options?: AgentCreateOptions) => ChatAgent>' is not assignable to type 'MockContext<(this: unknown, ...args: unknown[]) => unknown>'.
      Type '(chatId: string, callbacks: PilotCallbacks, options?: AgentCreateOptions) => ChatAgent' is not assignable to type '(this: unknown, ...args: unknown[]) => unknown>'.
        Types of parameters 'chatId' and 'args' are incompatible.
          Type 'unknown' is not assignable to type 'string'.
```

## 根本原因分析

在 `src/schedule/scheduler.test.ts` 第 92 行，`createScheduleAgentSpy` 变量的类型声明为：

```typescript
let createScheduleAgentSpy: ReturnType<typeof vi.spyOn>;
```

这会被 TypeScript 推断为泛型类型 `MockInstance<(this: unknown, ...args: unknown[]) => unknown>`。

但在第 102 行，实际赋值的类型是：

```typescript
createScheduleAgentSpy = vi.spyOn(AgentFactory, 'createScheduleAgent').mockReturnValue(mockAgent);
```

`vi.spyOn(AgentFactory, 'createScheduleAgent')` 返回的是更具体的类型：
`MockInstance<(chatId: string, callbacks: PilotCallbacks, options?: AgentCreateOptions) => ChatAgent>`

这两个类型不兼容，导致 TypeScript 编译错误。

## 修复建议

将第 92 行的类型声明改为更具体的类型：

```typescript
import type { MockInstance } from 'vitest';

let createScheduleAgentSpy: MockInstance<(chatId: string, callbacks: PilotCallbacks, options?: AgentCreateOptions) => ChatAgent>;
```

或者使用类型断言：

```typescript
createScheduleAgentSpy = vi.spyOn(AgentFactory, 'createScheduleAgent', 'get') as unknown as typeof createScheduleAgentSpy;
```

## 相关信息

- 触发提交: d1be1bf (Merge PR #724 - Agent 生命周期相关)
- 影响文件: `src/schedule/scheduler.test.ts`
- 相关 Issue: #711 (Agent 生命周期)

🤖 Generated by CI Monitor scheduled task

--- a/src/schedule/scheduler.test.ts
+++ b/src/schedule/scheduler.test.ts
@@ -15,7 +15,7 @@ import * as path from 'path';
 import * as os from 'os';
 import { Scheduler } from './scheduler.js';
 import { ScheduleManager } from './schedule-manager.js';
-import { AgentFactory } from '../agents/index.js';
+import { AgentFactory, type AgentCreateOptions } from '../agents/index.js';
 import type { ScheduledTask } from './index.js';
 import type { PilotCallbacks } from '../agents/pilot.js';
 import type { ChatAgent } from '../agents/types.js';
@@ -89,7 +89,7 @@ describe('Scheduler', () => {
   let mockCallbacks: PilotCallbacks;
   let testDir: string;
   let mockAgent: ChatAgent;
-  let createScheduleAgentSpy: MockInstance<(...args: unknown[]) => ChatAgent>;
+  let createScheduleAgentSpy: MockInstance<(chatId: string, callbacks: PilotCallbacks, options?: AgentCreateOptions) => ChatAgent>;
 
   beforeEach(async () => {
     testDir = path.join(os.tmpdir(), `scheduler-test-${Date.now()}-${Math.random().toString(36).slice(2)}`);

